modules/videopress/videopress.php

```diff
@@ -731,6 +732,32 @@ function videopress_shortcode_options( $options ) {
 		return $options;
 	}
 
+	/**
+	 * An override for the attachment url, which returns back the WPCOM videopress original url,
+	 * if it is set to the the objects metadata. this allows us to show the original uploaded
+	 * file on the WPCOM architecture, instead of the locally uplodaded file,
+	 * which doeasn't exist.
+	 *
+	 * @param string $url
+	 * @param int $post_id
+	 *
+	 * @return mixed
+	 */
+	public function update_attachment_url_for_videopress( $url, $post_id ) {
+
+		$post = get_post( $post_id );
+
+		if ( $post->post_mime_type === 'video/videopress' ) {
+			$meta = wp_get_attachment_metadata( $post->ID );
+
+			if ( isset( $meta['original']['url'] ) ) {
```

Comment:
What's your thinking on the `create_local_media_library_for_videopress_guid()` function -- should we update it so it just calls `xmlrpc_update_videopress_info()` instead of doing the work itself? https://github.com/Automattic/jetpack/blob/master/modules/videopress/utility-functions.php#L142